src/share/classes/com/sun/javatest/regtest/agent/JUnitRunner.java

```diff
@@ -91,4 +109,40 @@ public static void main(ClassLoader loader, String... args) throws Exception {
         }
     }
 
+    private static void runWithJUnitPlatform(Class<?> mainClass) throws Exception {
+        // https://junit.org/junit5/docs/current/user-guide/#launcher-api-execution
+        Thread.currentThread().setContextClassLoader(mainClass.getClassLoader());
+        try {
+
+            LauncherDiscoveryRequest request = LauncherDiscoveryRequestBuilder.request()
+                .selectors(DiscoverySelectors.selectClass(mainClass))
+                .build();
+
+            SummaryGeneratingListener summaryGeneratingListener = new SummaryGeneratingListener();
+
+            try (LauncherSession session = LauncherFactory.openSession()) {
+                Launcher launcher = session.getLauncher();
+                launcher.registerTestExecutionListeners(summaryGeneratingListener);
+                launcher.execute(request);
+            }
+
+            TestExecutionSummary summary = summaryGeneratingListener.getSummary();
+            if (summary.getTotalFailureCount() > 0) {
+                StringWriter sw = new StringWriter();
+                try (PrintWriter pw = new PrintWriter(sw)) {
+                    pw.println("JavaTest Message: JUnit Platform Failure(s): " + summary.getTotalFailureCount());
+                    pw.println();
+                    for (TestExecutionSummary.Failure failure : summary.getFailures()) {
+                        failure.getException().printStackTrace(pw);
+                    }
+                    summary.printTo(pw);
+                }
+                System.err.println(sw);
```

Comment:
printing to `System.err` may be an issue; this is happening on the test-side, isn't it?

Reply:
... I guess it might just mean we can't use JUnit Jupiter for some of the low-level core-libs tests

Reply:
It's doing the same as the JUnit 4 runner already did before. Thus, it shouldn't introduce new issues.